packages/sdk/client-services/src/packlets/spaces/notarization-plugin.ts

```diff
@@ -156,13 +205,32 @@ export class NotarizationPlugin implements CredentialProcessor {
 
     notarizeTask.schedule();
     this._extensionOpened.on(ctx, () => notarizeTask.schedule());
+  }
 
-    try {
-      await Promise.race([rejectOnDispose(ctx), allNotarized, errors.wait()]);
-      log('done');
-    } finally {
-      await ctx.dispose();
-    }
+  private _tryNotarizeCredentialsWithEdge(
+    ctx: Context,
+    client: EdgeHttpClient,
+    credentials: Credential[],
+    timeouts: NotarizationTimeouts & { jitter?: number },
+  ) {
+    const encodedCredentials = credentials.map((credential) => {
+      const binary = credentialCodec.encode(credential);
+      return Buffer.from(binary).toString('base64');
+    });
+    scheduleTask(ctx, async () => {
+      try {
+        await client.post(`/spaces/${this._spaceId}/notarization`, {
```

Comment:
im wondering if we could make every request it's own method on the client with apporpriate TS types